### PR TITLE
Shrink filter

### PR DESCRIPTION
**Pull request by mahmoudsamyhamed**
Added shrink filter

### CODE REVIEW

**Review by a7medev — APPROVED** [review]

File: main.cpp [hunk 1]
```diff
@@ -586,4 +593,51 @@ void mirrorImage()
...
+        isValidFactor = choice >= 1 && choice <= 3;
+    }
+
+    float factor = choice + 1;
+
+    for (int i = 0; i < SIZE; i += factor)
+    {
+        for (int j = 0; j < SIZE - factor + 1; j++)
+        {
+            int iFactor = ceil(float(i) * (1 / factor)), jFactor = ceil(float(j) * (1 / factor));
```
Comment (a7medev) [inline main.cpp:631]: Let's separate the declaration of `iFactor` and `jFactor` for readability.

I also feel like we can give the m different names like `scaledI` (or `scaledRow`) and `scaledJ` (or `scaledColumn`) (or something else that's more descriptive).

What do you think?

File: main.cpp [hunk 2]
```diff
@@ -586,4 +593,51 @@ void mirrorImage()
...
+            int newPixelValue = 0;
+
+            for (int k = 0; k < factor; k++)
+            {
+                newPixelValue += originalImage[i][j + k];
+            }
```
Comment (a7medev) [inline main.cpp:637]: I wonder what's the purpose of this piece of code. Is it just to average the removed pixels? If so, how would that be different from just ignoring the removed ones altogether and picking only one of them depending on the shrinking factor?

**Review by a7medev — CHANGES_REQUESTED** [review]

File: main.cpp (hunk 2 above)
Comment (mahmoudsamyhamed) [reply main.cpp:637]: They actually achieve the same thing but this makes the image look smoother.

File: main.cpp (hunk 1 above)
Comment (mahmoudsamyhamed) [reply main.cpp:631]: Yeah i agree i will change them right away.

**Review by a7medev — APPROVED** [review]
Let's go! 🚀

**Pull request merged**